src/main/java/net/snowflake/ingest/streaming/SnowflakeStreamingIngestChannel.java

```diff
@@ -11,11 +11,17 @@
 
 /**
  * A logical partition that represents a connection to a single Snowflake table, data will be
- * ingested into the channel, and then flush to Snowflake table periodically in the background. Note
- * that only one client (or thread) could write to a channel at a given time, so if there are
- * multiple clients (or multiple threads in the same client) try to ingest using the same channel,
- * the latest client (or thread) that opens the channel will win and all the other opened channels
- * will be invalid
+ * ingested into the channel, and then flushed to Snowflake table periodically in the background.
+ *
+ * <p>Channels are identified by their name and only one channel with the same name may ingest data
+ * at the same time. When a new channel is opened, all previously opened channels with the same name
+ * are invalidated (this applies for the table globally. not just in a single JVM). In order to
+ * ingest data from multiple threads/clients/applications, we recommend opening multiple channels,
+ * each with a different name. There is no limit on the number of channels that can be opened.
+ *
+ * <p>Thread safety note: Implementations of this interface are required to be thread safe, one
+ * instance of a channel can be used to ingest data from multiple threads. The recommended way,
+ * however, would be to open a new channel in every thread.
```

Comment:
I can see the multiple thread use case for certain application. I think we should recommend the per-channel thread but it is not strictly a requirement.

Reply:
@sfc-gh-tzhang There are frameworks where the users don't have control over which thread is executing their code because everything is handled by the framework (e.g. Akka). I wanted to point out that users don't have to be afraid to use a single channel in situations like that.

Reply:
@sfc-gh-tzhang Is this phrasing better?

> Implementations of this interface are required to be thread safe. We recommend that every thread ingests data via its own dedicated channel, although it is not a requirement.

Reply:
Thinking about this further I think we should just specify the thread activity and not make a recommendation. I'm thinking of queueing systems that offer "pop" semantics where different threads read from the same logical queue where sharding isn't supported.

Reply:
I should be more clear, the part I don't want to mention is to allow multiple threads inserting into the same channel **concurrently**, since there is no benefits on throughout as we have a internal lock anyway, it's ok if they want to guarantee ordering and use the same channel object from multiple threads.

Reply:
So should it just say 

> Thread safety note: Implementations of this interface are required to be thread safe.

and that's it? Everyone can draw their own conclusions 😄

Reply:
I would vote for adding a recommendation and what @sfc-gh-lsembera had above looks good to me, we might keep getting questions if we don't have any recommendations :)

Reply:
I like leaving it as `Thread safety note: Implementations of this interface are required to be thread safe.`

Reply:
I don't know if Javadocs are the place to provide application recommendations.

Reply:
I removed the recommendation